sky/skylet/providers/lambda_labs/lambda_utils.py

```diff
@@ -0,0 +1,161 @@
+"""Lambda Labs helper functions."""
+import os
+import json
+import requests
+
+CREDENTIALS_PATH = '~/.lambda_labs/lambda_keys'
+API_ENDPOINT = 'https://cloud.lambdalabs.com/api/v1'
+
+
+class LambdaLabsError(Exception):
+    pass
+
+
+class Metadata:
+    """Per-cluster metadata file."""
+
+    def __init__(self, path_prefix, cluster_name):
+        # TODO(ewzeng): Metadata file is not thread safe. This is fine for
+        # now since SkyPilot uses a per-cluster lock for ray-related
+        # operations, but this should be improved in the future.
+        self.path = os.path.expanduser(f'{path_prefix}-{cluster_name}')
+        self._metadata = {}
+        # In case parent directory does not exist
+        os.makedirs(os.path.dirname(self.path), exist_ok=True)
+        if os.path.exists(self.path):
+            with open(self.path, 'r') as f:
+                self._metadata = json.load(f)
+
+    def __getitem__(self, instance_id):
+        return self._metadata.get(instance_id)
+
+    def __setitem__(self, instance_id, value):
+        if value is None:
+            if instance_id in self._metadata:
+                self._metadata.pop(instance_id) # del entry
+            if (len(self._metadata) == 0 and
+                    os.path.exists(self.path)):
+                os.remove(self.path)
+                return
+        else:
+            self._metadata[instance_id] = value
+        with open(self.path, 'w') as f:
+            json.dump(self._metadata, f)
+
+
+def raise_lambda_error(response):
+    """Raise LambdaLabsError if appropriate. """
+    status_code = response.status_code
+    if status_code == 200:
+        return
+    if status_code == 429:
+        # https://docs.lambdalabs.com/cloud/rate-limiting/
+        raise LambdaLabsError('Your API requests are being rate limited.')
+    try:
+        resp_json = response.json()
+        code = resp_json['error']['code']
+        message = resp_json['error']['message']
+    except (KeyError, json.decoder.JSONDecodeError):
+        raise LambdaLabsError(f'Unexpected error. Status code: {status_code}')
+    raise LambdaLabsError(f'{code}: {message}')
+
+
+class LambdaLabsClient:
```

Comment:
Seems for every caller of the methods, we will always do `returnvalue.get('data', [])`. How about we have it in the functions instead?

Reply:
Good idea. PR updated.